src/main/java/com/linecorp/armeria/client/circuitbreaker/NonBlockingCircuitBreaker.java

```diff
@@ -186,6 +181,41 @@ private void logStateTransition(CircuitState circuitState, @Nullable EventCount
         }
     }
 
+    private void notifyStateChanged(CircuitState circuitState) {
+        config.listeners().forEach(listener -> {
+            try {
+                listener.onStateChanged(this, circuitState);
+            } catch (Throwable t) {
+                logger.warn("An error occured when notifying a state changed event", t);
+            }
+            try {
+                listener.onEventCountUpdated(this, EventCount.ZERO);
+            } catch (Throwable t) {
+                logger.warn("An error occured when notifying an EventCount updated event", t);
+            }
+        });
+    }
+
+    private void notifyCountUpdated(EventCount count) {
+        config.listeners().forEach(listener -> {
+            try {
+                listener.onEventCountUpdated(this, count);
+            } catch (Throwable t) {
+                logger.warn("An error occured when notifying an EventCount updated event", t);
+            }
+        });
+    }
+
+    private void notifyRequestRejected() {
+        config.listeners().forEach(listener -> {
+            try {
+                listener.onRequestRejected(this);
+            } catch (Throwable t) {
```

Comment:
This try-catch-logging is just to be consistent with the behavior of `notifyStateChanged`.